students/MzKhan/lesson03/mailroom.py

```diff
@@ -0,0 +1,153 @@
+    return donor_name
+
+def donation_prompt(name):
+    # This method prompts the user for the donation amount for the given donor.
+    # parm: list
+    # return: float
+    donation_amount = input("Please enter the donation amount for {} {}: "
+                            .format(*name))
+    return float(donation_amount)
+
+def display_list(donors_data=donors_list):
+    # This method prints all the donor's list for the user.
+    # parm: list    (kwarg)
+    print("\nOur generous donors: \n")
+    for donor in donors_data:
+        print(donor[0],donor[1])
+
+def find_donor(donor_name, donors_data = donors_list):
+    # This method checks if the given donor exists in the donor's list.
+    # If the donor does exist, it returns its index
+    # If the donor doesn't exist, it returns the index to add the new donor.
+    # parm: list
+    # parm: list    (kwarg)
+    # return: int
+    index = len(donors_data)
+    for donor in donors_data:
+        if (donor_name[0] == donor[0] and donor_name[1] == donor[1]):
+            index = donors_data.index(donor)
+    return index
+
+def email(donor_name, amount):
+    # This method generates the email for the given donor for his/her donation.
+    # parm: list
+    # parm: float
+    header = "\nDear {:} {:}".format(*donor_name)
+    body = "\nThank you so much for your generous donation of $ {:.2f}."
+    print(header, body.format(amount))
+
+def print_report(donors_data = donors_list):
+    # This method prints the donors' report in the formatted tabular form.
+    # parm: list    (kwarg)
+    title = "{:24} | {:12} | {:10} | {:16}"
+    print(title.format('Donor Name','Total Given','Num Gifts','Average Gift'))
+    strf_format = "{:12} {:12} $  {:10.2f} {:10d}   ${:10.2f}"
+    print(67*('-'))
+    report_list = calculate_total_gift(donors_data)
+    report_list.sort(key = custom_sort, reverse = True)
+    for donor in report_list:
+        print(strf_format.format(*donor))
+
+def calculate_total_gift(donors_data = donors_list):
+    # This method creates a new list from the global donors' list
+    # where each item of the list is a list of
+    # the donor's first name, last name, total donation, number of donations,
+    # and an average donation
+    # parm: list     (kwarg)
+    # return: list
+    new_list = []
+    for donor in donors_data:
+        first,last = donor[0],donor[1]
+        total_donation, num_of_gifts = sum(donor[2:]), len(donor)-2
+        avg_gift = total_donation/num_of_gifts
+        new_list.append([first,last,total_donation,num_of_gifts,avg_gift])
+    return new_list
+
+def custom_sort(a_list):
+    # This method returns the total donation amount at index = 2
+    # parm: list
+    # return: list[2]
+    return a_list[2]
+
+if __name__ == "__main__":
+
+    #Main program flow control and the interaction with the user.
+
+    while (True):
+        user_response = prompt_user()
+        if ( user_response == 3):
+            break
```

Comment:
hint: you can map quit option 3 to a function as well (this will be important in next revision).

your quit function can call `sys.exit()`